api/python/t4/packages.py

```diff
@@ -441,8 +441,15 @@ def fetch(self, dest):
         """
         # TODO: do this with improved parallelism? connections etc. could be reused
         nice_dest = fix_url(dest).rstrip('/')
+
+        file_list = []
         for logical_key, entry in self.walk():
-            entry.fetch('{}/{}'.format(nice_dest, quote(logical_key)))
+            physical_key = _to_singleton(entry.physical_keys)
+            new_physical_key = f'{nice_dest}/{logical_key}'
```

Comment:
Still need `quote(logical_key)` - it's a URL. Also, maybe call it `dest_url` or something like that; calling it a physical key seems weird at this point.